src/qtt/algorithms/allxy.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict, Any, List, Union, Optional
+import numpy as np
+import matplotlib.pyplot as plt
+from lmfit import Model
+
+from qcodes import DataSet, DataArray
+from qtt.utilities.visualization import plot_vertical_line
+
+
+def generate_allxy_combinations() -> List[Any]:
+    """ Generate all combinations of the AllXY sequence from Reed 2013 """
+    xymapping = {'I': 'I', 'x': 'X90', 'y': 'Y90', 'X': 'X180', 'Y': 'Y180'}
+    ground_state_rotations = ['II', 'XX', 'YY', 'XY', 'YX']
+    equator_state_rotations = ['xI', 'yI', 'xy', 'yx', 'xY', 'yX', 'Xy', 'Yx', 'xX', 'Xx', 'yY', 'Yy']
+    excited_state_rotations = ['XI', 'YI', 'xx', 'yy']
+    allxy_combinations_input = ground_state_rotations + equator_state_rotations + excited_state_rotations
+    allxy_combinations = [(xymapping[x[0]], xymapping[x[1]]) for x in allxy_combinations_input]
+
+    return allxy_combinations
+
+
+def allxy_model(indices: Union[float, np.ndarray], offset0: float, slope0: float, offset1: float, slope1: float, offset2: float, slope2: float) -> Union[float, np.ndarray]:
```

Comment:
You don't want to convert the arguments `slope0`, `slope1` and `slope2` into a single tuple argument `slopes`? Same for `offset0`, `offset1` and `offset2` into 'offsets'?

Reply:
Grouping parameters does not does work for the `lmfit` module. We could do a 
```
def _allxy_model(index, offset0, offset1, offset2, slope0, slope1, slope2):
   ....
   ...
   return calculated_value

def allxy_model(index, offsets, slopes):
   return _allxy_model(index, *offsets, *slopes)
```
In that way the public interface is `def allxy_model(index, offsets, slopes)`, but internally we can calculate with `lmfit` using `_allxy_model`. Alternatively we could do
```
def _allxy_model(index, offset0, offset1, offset2, slope0, slope1, slope2):
   ....
   ...
   return calculated_value

def allxy_model(index, parameters):
   return _allxy_model(index, *parameters)
```
In qtt we do both, e.g. compare https://github.com/QuTech-Delft/qtt/blob/dev/src/qtt/algorithms/functions.py#L13 or https://github.com/QuTech-Delft/qtt/blob/dev/src/qtt/algorithms/functions.py#L80

@CumulonimbusCalvus @QFer I don't really mind which convention to use, we can convert it either way. For this particular case I would pick either
```
def allxy_model(index, parameters): 
```
or
```
def _allxy_model(index, offset0, offset1, offset2, slope0, slope1, slope2)
```

Reply:
Ah, now I see why this is needed. The `parameters` option makes most sense to me, as long if the `parameters` argument is well documented.